app/src/main/java/com/zulip/android/ZulipActivity.java

```diff
@@ -54,6 +57,48 @@
         MessageListFragment.Listener {
 
     ZulipApp app;
+    List<Message> muted_topics;
+    @Override
+    public void addToList(Message message) {
+        muted_topics.add(message);
+    }
+
+    @Override
+    public void unMuteTopic(int StreamID, String subject) {
+        app.UnMuteTopic(StreamID, subject);
+        int added = 0;
+        for (Iterator<Message> messageIterator = muted_topics.iterator(); messageIterator.hasNext(); ) {
```

Comment:
A cleaner way to do this would be by implementing `equals` and `hashCode` on the `Message` object; this would enable you to be able to call `muted_topics.remove(message)`.